app/src/main/java/org/fossasia/openevent/general/auth/AuthService.kt

```diff
@@ -33,39 +30,31 @@ class AuthService(private val authApi: AuthApi,
 
     fun isLoggedIn() = authHolder.isLoggedIn()
 
-    fun logout() {
-        val userFlowable = userDao.getUser(authHolder.getId())
-        userFlowable.map {
-            Single.fromCallable {
+    fun logout(): Completable {
+        return Completable.fromAction({
+            var userFlowable = userDao.getUser(authHolder.getId())
+            userFlowable.map {
                 userDao.deleteUser(it)
-            }.subscribeOn(Schedulers.io())
-                    .observeOn(AndroidSchedulers.mainThread())
-                    .subscribe({
-                        Timber.d("User logged out!")
-                    }
-                    ) {
-                        Timber.e(it, "Error logging out user!")
-                    }
-        }
-        authHolder.token = null
+            }
+            authHolder.token = null
+        })
+
     }
 
 
-    fun getProfile(id: Long = authHolder.getId()): Flowable<User> {
-        val userFlowable = userDao.getUser(id)
-        userFlowable.switchIfEmpty{
-            authApi.getProfile(id)
-                    .map {
-                        userDao.insertUser(it)
-                    }
-                    .toFlowable()
-                    .flatMap {
-                        userFlowable
-                    }
+    fun getProfile(id: Long = authHolder.getId()): Single<User> {
+        val userSingle = userDao.getUser(id)
+        userSingle.onErrorResumeNext {
+            var profileSingle = authApi.getProfile(id)
+            profileSingle.map {
+                userDao.insertUser(it)
+            }
+            return@onErrorResumeNext profileSingle
```

Comment:
Return directly without storing in intermediate variable

Reply:
one map statement also has to be executed that's why intermediated variable is added

Reply:
Can be done with implicit returning as well